src/error.rs

```diff
@@ -0,0 +1,9 @@
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+/// The specific type of an error.
```

Comment:
This description looks vague to me. How about something like

```suggestion
/// The type of error that the library `collections` might return.
```

Reply:
I don't know. It's obvious that this this the error collections returns

Reply:
Something simpler if you prefer like 
```suggestion
/// Typed error for `collections`.
```

I guess I'm annoyed by `specific` mostly. What does this means?

Reply:
specific means it belongs to collections
it's the same as saying `Typed error for `collections` but writing `collections` for me it's pretty obvious . It's not for other crate.

I don't want spend so much time on it so I will change it